src/Compilers/CSharp/Test/Syntax/Diagnostics/DiagnosticTest.cs

```diff
@@ -2061,6 +2061,66 @@ public static void Main()
             AssertEx.Equal(CreateCompilationWithMscorlib(tree).GetDiagnostics(), CreateCompilationWithMscorlib(tree).GetDiagnostics());
         }
 
+        /// <summary>
+        /// Test that invalid type argument lists produce clean error messages
+        /// with minimal noise
+        /// </summary>
+        [WorkItem(7177, "https://github.com/dotnet/roslyn/issues/7177")]
+        [Fact]
+        public void InvalidTypeArgumentList()
+        {
+            var text = @"using System;
+public class A
+{
+    static void Main(string[] args)
+    {
+        // Invalid type arguments
+        object a1 = typeof(Action<0>);
+        object a2 = typeof(Action<static>);
+
+        // Valid type arguments
+        object a3 = typeof(Action<string>);
+        object a4 = typeof(Action<>);
+
+        // Invalid with multiple types
+        object a5 = typeof(Func<0,1>);
+        object a6 = typeof(Func<0,bool>);
+        object a7 = typeof(Func<static,bool>);
+
+        // Valid with multiple types
+        object a8 = typeof(Func<string,bool>);
+        object a9 = typeof(Func<,>);
+
+        // Invalid with nested types
+        object a10 = typeof(Action<Action<0>>);
+        object a11 = typeof(Action<Action<static>>);
+        object a12 = typeof(Action<Action<>>);
+
+        // Valid with nested types
+        object a13 = typeof(Action<Action<string>>);
+    }
+}";
+
+            var comp = DiagnosticsUtils.VerifyErrorsAndGetCompilationWithMscorlib(text,
+                // Case a1
```

Comment:
Please, use baseline style as in PragmaWarning_IdentifierErrorCodesMustMatchExactly1 test. I.e. the error number, message text and corresponding source code is included as a comment. I suggest switching to using VerifyDiagnostics helper, this is the preferred helper to verify expected diagnostics, only two tests in this file are still using VerifyErrorsAndGetCompilationWithMscorlib helper. Also, if you use VerifyDiagnostics helper and run the test without supplying any arguments to it, the test failure output will contain baseline that you can simply copy  and paste as the argument list for VerifyDiagnostics.